src/main/java/de/tum/in/www1/artemis/web/rest/RepositoryResource.java

```diff
@@ -161,10 +165,66 @@ public RepositoryResource(UserService userService, ParticipationService particip
     }
 
     /**
-     * DELETE /repository/{participationId}/file: Delete the file
+     * POST /repository/{participationId}/folder: Create new folder
      *
      * @param participationId Participation ID
      * @param filename
+     * @param request
+     * @return
+     * @throws IOException
+     */
+    @PostMapping(value = "/repository/{participationId}/folder", produces = MediaType.APPLICATION_JSON_VALUE)
```

Comment:
please add security checks so that not everyone can invoke this REST call

Reply:
Ass discussed we check the participation of the user in the endpoint (is owner or at least TA or Instructor).